integrations/Braze/browser.js

```diff
@@ -172,28 +154,106 @@ class Braze {
       "gender",
       "lastname",
       "phone",
-      "facebook",
-      "twitter",
-      "first_name",
-      "last_name",
       "dob",
       "external_id",
       "country",
       "home_city",
       "bio",
-      "gender",
-      "phone",
       "email_subscribe",
       "push_subscribe",
     ];
+    // function set Address
+    function setAddress() {
+      window.braze.getUser().setCountry(address.country);
+      window.braze.getUser().setHomeCity(address.city);
+    }
+    // function set Birthday
+    function setBirthday() {
+      window.braze
+        .getUser()
+        .setDateOfBirth(
+          birthday.getUTCFullYear(),
+          birthday.getUTCMonth() + 1,
+          birthday.getUTCDate()
+        );
+    }
+    // function set Email
+    function setEmail() {
+      window.braze.getUser().setEmail(email);
+    }
+    // function set firstName
+    function setFirstName() {
+      window.braze.getUser().setFirstName(firstname);
+    }
+    // function set gender
+    function setGender(genderName) {
+      window.braze.getUser().setGender(genderName);
+    }
+    // function set lastName
+    function setLastName() {
+      window.braze.getUser().setLastName(lastname);
+    }
+    function setPhone() {
+      window.braze.getUser().setPhoneNumber(phone);
+    }
+
+    // deep clone the traits object
+    let traits = {};
+    if (rudderElement.message?.context?.traits) {
+      traits = cloneDeep(rudderElement.message.context.traits);
+    }
 
     reserved.forEach((element) => {
       delete traits[element];
     });
 
-    Object.keys(traits).forEach((key) => {
-      window.braze.getUser().setCustomUserAttribute(key, traits[key]);
-    });
+    if (
+      this.supportDedup &&
+      this.previousPayload !== null &&
+      userId &&
+      userId === this.previousPayload.rudderElement?.message?.userId
+    ) {
+      const prevMessage = this.previousPayload.rudderElement?.message;
+      const prevTraits = prevMessage?.context?.traits;
+      const prevAddress = prevTraits?.address;
+      const prevBirthday = prevTraits?.birthday || prevTraits?.dob;
+      const prevEmail = prevTraits?.email;
+      const prevFirstname = prevTraits?.firstname || prevTraits?.firstName;
+      const prevGender = prevTraits?.gender;
+      const prevLastname = prevTraits?.lastname || prevTraits?.lastName;
+      const prevPhone = prevTraits?.phone;
+
+      if (address && !isEqual(address, prevAddress)) setAddress();
```

Comment:
address is an object, we have to do check using custom comparator including all the address fields

Reply:
Yes, we are using lodash isEqual which will deeply check all fields and return boolean accordingly.